sway/tree/view.c

```diff
@@ -1047,3 +1053,27 @@ bool view_is_visible(struct sway_view *view) {
 	}
 	return true;
 }
+
+void view_set_urgent(struct sway_view *view, bool enable) {
+	if (enable) {
+		struct sway_seat *seat = input_manager_current_seat(input_manager);
+		if (seat_get_focus(seat) == view->swayc) {
+			return;
+		}
+		clock_gettime(CLOCK_MONOTONIC, &view->urgent);
+	} else {
+		view->urgent = (struct timespec){ 0 };
+		if (view->urgent_timer) {
+			wl_event_source_remove(view->urgent_timer);
+			view->urgent_timer = NULL;
+		}
+	}
+	container_damage_whole(view->swayc);
+
+	struct sway_container *ws = container_parent(view->swayc, C_WORKSPACE);
+	ipc_event_workspace(ws, NULL, "urgent");
```

Comment:
The arguments should be `NULL, ws, "urgent"` since the prototype has the old workspace first, then the current ("new") workspace. Also, do you send an event for the window too?